web/src/features/fba/components/infoPanel/AdvisoryText.tsx

```diff
@@ -147,68 +105,63 @@ const AdvisoryText = ({
-    setFilteredFireCentreHFIFuelStats(newFilteredFireCentreHFIFuelStats)
-  }, [fireCentreHFIFuelStats, selectedFireZoneUnit])
-
-  useEffect(() => {
+  const filteredFireCentreHFIFuelStats = useFilteredFireCentreHFIFuelStats()
+
+  // derived state
+  const selectedFilteredZoneUnitFuelStats = useMemo<FireZoneHFIStats>(() => {
     if (
       isUndefined(filteredFireCentreHFIFuelStats) ||
       isEmpty(filteredFireCentreHFIFuelStats) ||
       isUndefined(selectedFireCenter) ||
       isUndefined(selectedFireZoneUnit)
     ) {
-      setSelectedFireZoneUnitTopFuels([])
-      return
+      return { fuel_area_stats: [], min_wind_stats: [] }
     }
+    const allFilteredZoneUnitFuelStats = filteredFireCentreHFIFuelStats[selectedFireCenter.name]
+    return (
+      allFilteredZoneUnitFuelStats?.[selectedFireZoneUnit.fire_shape_id] ?? {
+        fuel_area_stats: [],
+        min_wind_stats: []
+      }
+    )
+  }, [filteredFireCentreHFIFuelStats, selectedFireCenter, selectedFireZoneUnit])
 
-    const allFilteredZoneUnitFuelStats = filteredFireCentreHFIFuelStats?.[selectedFireCenter.name]
-    const selectedFilteredZoneUnitFuelStats = allFilteredZoneUnitFuelStats?.[selectedFireZoneUnit.fire_shape_id] ?? {
-      fuel_area_stats: [],
-      min_wind_stats: []
-    }
-    const topFuels = getTopFuelsByArea(selectedFilteredZoneUnitFuelStats, forDate)
-    setSelectedFireZoneUnitTopFuels(topFuels)
-  }, [filteredFireCentreHFIFuelStats])
-
-  useEffect(() => {
-    const { minStartTime, maxEndTime, duration } = getMinStartAndMaxEndTime(selectedFireZoneUnitTopFuels)
-    setMinStartTime(minStartTime)
-    setMaxEndTime(maxEndTime)
-    setCriticalHoursDuration(duration)
+  const selectedFireZoneUnitTopFuels = useMemo<FireZoneFuelStats[]>(() => {
+    return getTopFuelsByArea(selectedFilteredZoneUnitFuelStats, forDate)
+  }, [selectedFilteredZoneUnitFuelStats, forDate])
+
+  const highHFIFuelsByProportion = useMemo<FireZoneFuelStats[]>(() => {
+    return getTopFuelsByProportion(selectedFilteredZoneUnitFuelStats.fuel_area_stats)
+  }, [selectedFilteredZoneUnitFuelStats])
+
+  const selectedFireZoneUnitMinWindSpeeds = selectedFilteredZoneUnitFuelStats.min_wind_stats
+
+  const {
+    minStartTime,
+    maxEndTime,
+    duration: criticalHoursDuration
+  } = useMemo(() => {
+    return getMinStartAndMaxEndTime(selectedFireZoneUnitTopFuels)
   }, [selectedFireZoneUnitTopFuels])
 
+  const getZoneStatus = () => {
+    if (selectedFireCenter) {
+      const fireCenterSummary = provincialSummary[selectedFireCenter.name]
+      const fireZoneUnitInfos = fireCenterSummary?.filter(
+        fc => fc.fire_shape_id === selectedFireZoneUnit?.fire_shape_id
+      )
+      const zoneStatus = calculateStatusText(fireZoneUnitInfos, advisoryThreshold)
+      return zoneStatus
+    }
+  }
+
+  const zoneStatus = useMemo(
+    () => getZoneStatus(),
+    [selectedFireCenter, selectedFireZoneUnit, provincialSummary, advisoryThreshold]
```

Comment:
I think the `advisoryThreshold` dependency can be removed as we have no mechanism in place for users to change this.